projects/ontimize-web-ngx/src/lib/validators/o-validators.ts

```diff
@@ -82,4 +82,35 @@ export class OValidators {
     return undefined;
   }
 
+  /**
+   * Patterns validator
+   * @param regex
+   * @param key
+   * @returns validator
+   *
+   */
+  public static patternValidator(regex: RegExp, key: string): ValidatorFn {
+
+    let validator: ValidatorFn = (control: FormControl): { [key: string]: any } => {
+      if (!control.value) {
+        // if control is empty return no error
+        return undefined;
+      }
+
+      // test the value of the control against the regexp supplied
+      const valid = regex.test(control.value);
```

Comment:
Maybe it would be necessary to check that _regex_ arg is not undefined for avoiding exceptions